### PR TITLE
perf(cbv2): default-on prefill stack — solo stripe, prompt narrowing, packed prefill (Qwen3.6), mean-TTFT cap

### PR DESCRIPTION
**Pull request by Gajesh2007**
## What

CBv2 prefill stack, enabled by default, all levers individually escapable:

1. **Solo-prefill stripe** (default 2048): one live text request widens its chunk 512→2048. Escape: `DARKBLOOM_CBV2_SOLO_PREFILL_STRIPE=0`.
2. **Recurrent prompt narrowing** (Qwen LM head): intermediate chunks return a `[B,1]` handle, the frontier projects one row instead of `[1,512,248320]` (242.5 MiB/chunk). Escape: `DARKBLOOM_CBV2_PREFILL_NARROWING=0`.
3. **Packed prefill for Qwen3.6**: equal-length chunks from B requests run as one `[B,L]` forward (one weight stream per cohort; per-row recurrent state). Text-only v1.
4. **Mean-TTFT prefill serialization** (`DARKBLOOM_CBV2_MAX_PARTIAL_PREFILLS`, opt-in).

Companion mlx-swift-lm PR carries the engine/model changes; this PR wires env/defaults + pin.

## Measured (M4 Max, Qwen3.6 35B-A3B, HPM, 8K, 3-iter medians)

| config | TTFT | prefill tok/s |
|---|---:|---:|
| 2026-08-18 baseline (`=1`, 512) | 6406.8 ms | 1,279 |
| trust (#638) + stripe | 5306.3 ms | 1,544 |
| **+ narrowing (this stack, solo)** | **4636.9 ms** | **~1,766 (+38%)** |
| multi-request 4x8K aggregate (was 1,312) | — | **1,479-1,537 (+13-17%)** |

`outputsMatchBurst=true` on every arrival pattern (token-checksum parity across all levers). Tests: 16 feature + 166 regression suites green. Solo stripe regresses ~12% under Low Power Mode — throttled machines should export the stripe escape (documented in `docs/reports/2026-08-19-solo-prefill-stripe-experiment.md`).

## Behavior

```mermaid
flowchart LR
  subgraph Before
    A1[solo 8K prompt] --> B1[16x512 chunks - full logits each]
    A2[4x8K burst] --> B2[per-row forwards - 4x weight streams - all TTFTs = makespan]
  end
  subgraph After
    C1[solo 8K prompt] --> D1[4x2048 stripes - frontier-row-only projection]
    C2[4x8K burst] --> D2["packed [4,512] cohorts - one weight stream - +13-17% aggregate"]
    D1 --> E1[company arrives -> next plan is plain 512s]
  end
```

## Code

```mermaid
flowchart LR
  subgraph Before
    P1[plan: chunk=min(rem,512,budget)] --> F1[targetForward full logits] --> N1[engine slice]
  end
  subgraph After
    P2[plan: solo gate / partial-prefill cap] --> F2["targetForward(requirement:)"]
    F2 -->|CBv2RecurrentPrefillSteppableModel| M2[model narrows: B,1 handle or one-row LM head]
    F2 -->|fallback| N2[engine slice - byte-old]
    G2[packed executor] -->|recurrent group| F2
  end
```

---
<a href="https://app.blacksmith.sh/Layr-Labs/codesmith/d-inference/pr/646"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with [code]smith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1789777191&installation_model_id=21733&pr_number=646&repository=Layr-Labs%2Fd-inference&return_to=https%3A%2F%2Fgithub.com%2FLayr-Labs%2Fd-inference%2Fpull%2F646&signature=d2a910b69740df28e26094141a9f01dbc352c397062db2e637c1630da9fd4a7d"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with [code]smith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith-bot</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by Gajesh2007** [comment]
All four Codex findings addressed in 58dce8638 (+ companion mlx-swift-lm a3fed8c):
- **P1 paged lockstep test**: now pins `maxPrefillChunk == max(prefillChunkSize, stripe)` and asserts the 2048 default — verified green (`pagedPoolLocksStepWithSchedulerConfig`).
- **P2 injected environment**: stripe + partial-prefill knobs read the injected `environment` param, not the host process — synthetic-env callers and the `=0` escape behave.
- **P2 schema**: scheduler-prefill `currentSchemaVersion` 2→3 (posture field documented); wrapper validator updated to expect 3.
- **P2/P1 report**: stripe documented **default-ON** with the Low-Power-Mode escape guidance, canonical implementation citations added.

**Comment by Gajesh2007** [comment]
Round-2 findings fixed in $(git rev-parse --short HEAD):
- **P1 baseline invalidation**: wrapper `SCHEMA_VERSION` 5→6 with a documented rationale — the exact-match `validate_schema_version_pin` now refuses pre-flip baselines outright, so the default 2048-stripe posture can never read as a code delta. Chose the schema bump over a nested-payload check because the flip also changes arrival-invariance behavior (packed prefill), not just scheduler-prefill.
- **P2 fixtures**: `scheduler_payload` emits schema 3 + the posture field; `make benchmark-wrapper-test` green (62 tests).
- **P1 citations**: report's canonical-implementation section now carries file:line ranges for every behavior claim (gate, shrink-retries, cap fail-open, seam, packed cohort, defaults, lockstep), anchored to the current commits.

**Review by 0xkydo — DISMISSED** [review]

**Review by 0xkydo — APPROVED** [review]
